app/src/main/java/org/p2p/wallet/jupiter/repository/tokens/JupiterSwapTokensRemoteRepository.kt

```diff
@@ -146,4 +152,12 @@ internal class JupiterSwapTokensRemoteRepository(
     private fun Sequence<JupiterSwapToken>.filterTokensByAvailability(): Sequence<JupiterSwapToken> {
         return filter { it.tokenExtensions.isTokenCellVisibleOnWalletScreen != false }
     }
+
+    override suspend fun findTokenBySymbol(symbol: String): JupiterSwapToken? {
+        return getTokens().firstOrNull { it.tokenSymbol.equals(symbol, ignoreCase = true) }
+    }
```

Comment:
this is too expensive since getTokens() returns more than 2k elements, in first iteration it can be minimally optimized using an in-memory cache